src/plugins/lua/external_services.lua

```diff
@@ -181,12 +181,19 @@ local function add_scanner_rule(sym, opts)
       fun.each(function(p)
         local content = p:get_content()
         if content and #content > 0 then
-          cfg.check(task, content, p:get_digest(), rule)
+          local in_fname = p:get_filename()
+          local in_type, in_stype = p:get_detected_type()
+          local part_info = {in_fname,in_type,in_stype}
+          cfg.check(task, content, p:get_digest(), rule, part_info)
         end
       end, common.check_parts_match(task, rule))
 
     else
-      cfg.check(task, task:get_content(), task:get_digest(), rule)
+      local in_fname = "mail"
+      local in_type = "application"
+      local in_stype = "octet-stream"
+      local part_info = {in_fname,in_type,in_stype}
+      cfg.check(task, task:get_content(), task:get_digest(), rule, part_info)
```

Comment:
If we don't have part, then we must just pass `nil` here and reconstruct the necessary information afterwards.

Reply:
Maybe if it's nil we can stick with the already available req_fake_url? Otherwise, I am not sure how I can reconstruct the missing information in the lua_scanner itself... May you can point me in the right direction?